App/ViewModels/Address/AddressesViewModel.cs

```diff
@@ -85,28 +66,34 @@ private async Task ShowBookingView()
 
         private Task<bool> InflateWaypointsModel()
         {
-            Model.Waypoints.Points.Add(FromDomainAddress);
-            Model.Waypoints.Points.Add(ToDomainAddress);
+            if (SelectedFromAddress.Value == null || SelectedToAddressess.Value == null)
+            {
+                return Task.FromResult(false);
+            }
+
+            Model.Waypoints.Points.Add(SelectedFromAddress.Value);
+            Model.Waypoints.Points.Add(SelectedToAddressess.Value);
+            Model.Waypoints.Comment = Comment.Value;

```

Comment:
More appropriate names are StartAddress and EndAddress, because in the route can be middle's address points.